libs/community/langchain_community/document_loaders/parsers/images.py

```diff
@@ -0,0 +1,149 @@
+from langchain_community.document_loaders.blob_loaders import Blob
+
+logger = logging.getLogger(__name__)
+
+
+class ImageBlobParser(BaseBlobParser):
+    def __init__(
+        self,
+        *,
+        format: Literal["text", "markdown", "html"] = "text",
+    ):
+        self.format = format
+
+    @abstractmethod
+    def _analyze_image(self, img: "Image") -> str:
+        pass
+
+    def lazy_parse(self, blob: Blob) -> Iterator[Document]:
+        try:
+            from PIL import Image as Img
+
+            with blob.as_bytes_io() as buf:
+                img = Img.open(buf)
+                content = self._analyze_image(img)
+                if content:
+                    if self.format == "markdown":
+                        content = content.replace("]", r"\\]")
+                        content = f"![{content}](.)"
+                    elif self.format == "html":
+                        content = f'<img alt="{html.escape(content, quote=True)}" />'
+                logger.debug("Image text: %s", content.replace("\n", "\\n"))
+                yield Document(
+                    page_content=content,
+                    metadata={"source": blob.source},
+                )
+        except ImportError:
+            raise ImportError(
+                "`rapidocr-onnxruntime` package not found, please install it with "
+                "`pip install Pillow`"
+            )
+
+
+class RapidOCRBlobParser(ImageBlobParser):
+    def __init__(
+        self,
+        *,
+        format: Literal["text", "markdown", "html"] = "text",
+    ):
+        super().__init__(format=format)
+        self.ocr = None
+
+    def _analyze_image(self, img: "Image") -> str:
+        if not self.ocr:
+            try:
+                from rapidocr_onnxruntime import RapidOCR
+
+                self.ocr = RapidOCR()
+            except ImportError:
+                raise ImportError(
+                    "`rapidocr-onnxruntime` package not found, please install it with "
+                    "`pip install rapidocr-onnxruntime`"
+                )
+        ocr_result, _ = self.ocr(np.array(img))  # type: ignore
+        content = ""
+        if ocr_result:
+            content = ("\n".join([text[1] for text in ocr_result])).strip()
+        return content
+
+
+class TesseractBlobParser(ImageBlobParser):
+    def __init__(
+        self,
+        *,
+        format: Literal["text", "markdown", "html"] = "text",
+        langs: list[str] = ["eng"],
+    ):
+        super().__init__(format=format)
+        self.langs = langs
+
+    def _analyze_image(self, img: "Image") -> str:
+        try:
+            import pytesseract
+        except ImportError:
+            raise ImportError(
+                "`pytesseract` package not found, please install it with "
+                "`pip install pytesseract`"
+            )
+        return pytesseract.image_to_string(img, lang="+".join(self.langs)).strip()
+
+
+_prompt_images_to_description = (
```

Comment:
nit: style to follow google conventions a bit more closely

```suggestion
_PROMPT_IMAGES_TO_DESCRIPTION  = (
```